tasks/Module 3/task_2_test.py

```diff
@@ -0,0 +1,48 @@
+"""Task 2 of Module 3."""
+
+import re
+import pytest
+
+
+def my_splitter(to_split, separator=None):
+    """Create personal str.split() implementation with UnitTest."""
+    if separator is None:
+        split_list_regex = re.compile(r'[^\s]+')
+        return split_list_regex.findall(to_split)
+
+    split_list = []
+    separator_len = len(separator)
+
+    for i in to_split:
```

Comment:
The PyLint's complains with R1710 error because you have `return` statement only under the condition. To linter it means that your function returns something (list of strings) conditionally and in the other case it implicitly returns `None` and it does not like it. You can take a look at [Python style guide here](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) and search for "Be consistent in return statements"

Of course we are smarter than PyLint and we know that it will never return `None`. Direct solution to this is add explicit `return None` at the end of the function, **but** please don't do it for now, because we can do slightly better by applying some changes described below :)

So, two thoughts regarding this algorithm.

1. The `for` loop may be not the best choice here. It basically says: "for every character in input string do blah blah blah", but your code does not perform operations per character. Instead it works until it reach `else` clause and then returns. It cases where you want to repeat something if condition is met, the better choice is `while` loop.
2. In `else` clause this slicing `to_split[:len(to_split)]` is not necessary, because if `to_split` does not contain separator, then it means that `to_split` is the last word. Actually it is worth noticing that we want to execute this code unconditionally with every function call - we always have this last word left after the last occurrence of separator. 
3. This relates to PyLint issue. We can pull this return outside the loop. In such way, loop will be responsible for collecting words and will end when we have all words on `separated_word` list.

Having in mind points above you can try to rewrite your algorithm using `while` loop. You can start with `while separator in to_split:` or `while to_split:` (you can use strings in conditions just like this - empty string is False, nonempty is True in this context).

Reply:
@musk-ox thanks for advices!! Now code looks more simple and readeble
i've replaced `for loop` with `while`, removed main `if condition` and replaced `else` with new `if` that does have reasonable condition